src/cpp/shared_core/include/shared_core/system/User.hpp

```diff
@@ -131,6 +138,42 @@ class User
     */
    static Error getUserFromIdentifier(UidType in_userId, User& out_user);
 
+   /**
+    * @brief Overloaded assignment operator.
+    *
+    * @param in_other   The user to copy to this one.
+    *
+    * @return This user.
+    */
+   User& operator=(const User& in_other);
+
+   /**
+    * @brief Overloaded move operator.
+    *
+    * @param in_other   The user to move to this one.
+    *
+    * @return This user.
+    */
+   User& operator=(User&& in_other) noexcept;
```

Comment:
Similar question re: using compiler default.

Reply:
The compiler won't generate a default because `User::User(const User&)` and `User& User::operator=(const User&)` are implemented.

Reply:
Sorry, I meant using `= default` to force the compiler to generate the default version.

Reply:
Oh, that's a good point! Thanks!